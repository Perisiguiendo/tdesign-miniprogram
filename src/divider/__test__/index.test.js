import simulate from 'miniprogram-simulate';
import path from 'path';

// 文本位置
const textAlign = ['left', 'center', 'right'];

describe('Divider', () => {
  const divider = simulate.load(path.resolve(__dirname, `../divider`), 't-divider', {
    less: true,
    rootPath: path.resolve(__dirname, '../..'),
  });

  textAlign.forEach((align) => {
    it(`:align ${align} render correctly`, () => {
      const id = simulate.load({
        template: `<t-divider class="divider-align"></t-divider>`,
        data: {
          content: '文字信息',
          align,
        },
        usingComponents: {
          't-divider': divider,
        },
      });
      const comp = simulate.render(id);
      comp.attach(document.createElement('parent-wrapper'));

      expect(comp.toJSON()).toMatchSnapshot();
    });
  });

  it(`:content render correctly`, () => {
    const id = simulate.load({
      template: `<t-divider></t-divider>`,
      usingComponents: {
        't-divider': divider,
      },
    });
    const comp = simulate.render(id);
    comp.setData({
      content: '文字信息',
    });
    comp.attach(document.createElement('parent-wrapper'));

    expect(comp.toJSON()).toMatchSnapshot();
  });

  it(':dashed render correctly', () => {
    const id = simulate.load({
      template: `<t-divider></t-divider>`,
      usingComponents: {
        't-divider': divider,
      },
    });
    const comp = simulate.render(id);
    comp.setData({
      content: '文字信息',
      dashed: true,
    });
    comp.attach(document.createElement('parent-wrapper'));

    expect(comp.toJSON()).toMatchSnapshot();
  });

  it(':layout horizontal render correctly', () => {
    const id = simulate.load({
      template: `<t-divider></t-divider>`,
      usingComponents: {
        't-divider': divider,
      },
    });
    const comp = simulate.render(id);
    comp.setData({
      layout: 'horizontal',
      content: '文字信息',
      dashed: true,
    });
    comp.attach(document.createElement('parent-wrapper'));

    expect(comp.toJSON()).toMatchSnapshot();
  });

  it(':layout vertical render correctly', () => {
    const id = simulate.load({
      template: `<t-divider></t-divider>`,
      usingComponents: {
        't-divider': divider,
      },
    });
    const comp = simulate.render(id);
    comp.setData({
      layout: 'vertical',
      content: '文字信息',
    });
    comp.attach(document.createElement('parent-wrapper'));

    expect(comp.toJSON()).toMatchSnapshot();
  });

  it(':line-color render correctly', () => {
    const id = simulate.load({
      template: `<t-divider></t-divider>`,
      usingComponents: {
        't-divider': divider,
      },
    });
    const comp = simulate.render(id);
    comp.setData({
      lineColor: '#eee',
      content: '文字信息',
    });
    comp.attach(document.createElement('parent-wrapper'));

    expect(comp.toJSON()).toMatchSnapshot();
  });
});
