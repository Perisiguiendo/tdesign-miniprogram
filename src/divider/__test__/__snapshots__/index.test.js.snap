// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider :align center render correctly 1`] = `
<main>
  <t-divider
    class="divider-align"
  >
    <wx-view
      class=""
    >
      <wx-view
        class="t-class t-divider t-divider--horizontal t-divider--center  "
        style=""
      >
        <wx-view
          class="t-class-content t-divider__content"
        />
      </wx-view>
    </wx-view>
  </t-divider>
</main>
`;

exports[`Divider :align left render correctly 1`] = `
<main>
  <t-divider
    class="divider-align"
  >
    <wx-view
      class=""
    >
      <wx-view
        class="t-class t-divider t-divider--horizontal t-divider--center  "
        style=""
      >
        <wx-view
          class="t-class-content t-divider__content"
        />
      </wx-view>
    </wx-view>
  </t-divider>
</main>
`;

exports[`Divider :align right render correctly 1`] = `
<main>
  <t-divider
    class="divider-align"
  >
    <wx-view
      class=""
    >
      <wx-view
        class="t-class t-divider t-divider--horizontal t-divider--center  "
        style=""
      >
        <wx-view
          class="t-class-content t-divider__content"
        />
      </wx-view>
    </wx-view>
  </t-divider>
</main>
`;

exports[`Divider :content render correctly 1`] = `
<main>
  <t-divider>
    <wx-view
      class=""
    >
      <wx-view
        class="t-class t-divider t-divider--horizontal t-divider--center  "
        style=""
      >
        <wx-view
          class="t-class-content t-divider__content"
        />
      </wx-view>
    </wx-view>
  </t-divider>
</main>
`;

exports[`Divider :dashed render correctly 1`] = `
<main>
  <t-divider>
    <wx-view
      class=""
    >
      <wx-view
        class="t-class t-divider t-divider--horizontal t-divider--center  "
        style=""
      >
        <wx-view
          class="t-class-content t-divider__content"
        />
      </wx-view>
    </wx-view>
  </t-divider>
</main>
`;

exports[`Divider :layout horizontal render correctly 1`] = `
<main>
  <t-divider>
    <wx-view
      class=""
    >
      <wx-view
        class="t-class t-divider t-divider--horizontal t-divider--center  "
        style=""
      >
        <wx-view
          class="t-class-content t-divider__content"
        />
      </wx-view>
    </wx-view>
  </t-divider>
</main>
`;

exports[`Divider :layout vertical render correctly 1`] = `
<main>
  <t-divider>
    <wx-view
      class=""
    >
      <wx-view
        class="t-class t-divider t-divider--horizontal t-divider--center  "
        style=""
      >
        <wx-view
          class="t-class-content t-divider__content"
        />
      </wx-view>
    </wx-view>
  </t-divider>
</main>
`;

exports[`Divider :line-color render correctly 1`] = `
<main>
  <t-divider>
    <wx-view
      class=""
    >
      <wx-view
        class="t-class t-divider t-divider--horizontal t-divider--center  "
        style=""
      >
        <wx-view
          class="t-class-content t-divider__content"
        />
      </wx-view>
    </wx-view>
  </t-divider>
</main>
`;
